server/src/main/java/org/opensearch/action/admin/cluster/remotestore/repository/RemoteStoreRepositoryRegistrationHelper.java

```diff
@@ -0,0 +1,364 @@
+    }
+
+    /**
+     * During node join request this method will validate if the joining node has the same remote store node
+     * attributes as of the existing node in the cluster.
+     *
+     * TODO: The below check is valid till we support migration, once we start supporting migration a remote
+     *       store node will be able to join a non remote store cluster and vice versa. #7986
+     */
+    public static boolean compareRemoteStoreNodeAttributes(DiscoveryNode joiningNode, DiscoveryNode existingNode) {
+        String segmentRepositoryName = validateAttributeNonNull(existingNode, REMOTE_STORE_SEGMENT_REPOSITORY_NAME_ATTRIBUTE_KEY);
+        String translogRepositoryName = validateAttributeNonNull(existingNode, REMOTE_STORE_TRANSLOG_REPOSITORY_NAME_ATTRIBUTE_KEY);
+
+        compareAttribute(joiningNode, existingNode, REMOTE_STORE_SEGMENT_REPOSITORY_NAME_ATTRIBUTE_KEY);
+        compareAttribute(
+            joiningNode,
+            existingNode,
+            String.format(Locale.getDefault(), REMOTE_STORE_REPOSITORY_TYPE_ATTRIBUTE_KEY_FORMAT, segmentRepositoryName)
+        );
+        compareSettingsAttributes(
+            joiningNode,
+            existingNode,
+            String.format(Locale.getDefault(), REMOTE_STORE_REPOSITORY_SETTINGS_ATTRIBUTE_KEY_PREFIX, segmentRepositoryName)
+        );
+        compareAttribute(joiningNode, existingNode, REMOTE_STORE_TRANSLOG_REPOSITORY_NAME_ATTRIBUTE_KEY);
+        compareAttribute(
+            joiningNode,
+            existingNode,
+            String.format(Locale.getDefault(), REMOTE_STORE_REPOSITORY_TYPE_ATTRIBUTE_KEY_FORMAT, translogRepositoryName)
+        );
+        compareSettingsAttributes(
+            joiningNode,
+            existingNode,
+            String.format(Locale.getDefault(), REMOTE_STORE_REPOSITORY_SETTINGS_ATTRIBUTE_KEY_PREFIX, translogRepositoryName)
+        );
+
+        return true;
+    }
+
+    private static Settings buildSettings(DiscoveryNode node, String settingsAttributePrefix) {
+        Settings.Builder settings = Settings.builder();
+        node.getAttributes()
+            .entrySet()
+            .stream()
+            .filter(entry -> entry.getKey().startsWith(settingsAttributePrefix))
+            .forEach(entry -> settings.put(entry.getKey().replace(settingsAttributePrefix, ""), entry.getValue()));
+        return settings.build();
+    }
+
+    // TODO: Add logic to mark these repository as System Repository once thats merged.
+    // Visible For testing
+    public static RepositoryMetadata buildRepositoryMetadata(DiscoveryNode node, String name) {
+        String type = node.getAttributes().get(String.format(Locale.getDefault(), REMOTE_STORE_REPOSITORY_TYPE_ATTRIBUTE_KEY_FORMAT, name));
+        String settingsAttributePrefix = String.format(Locale.getDefault(), REMOTE_STORE_REPOSITORY_SETTINGS_ATTRIBUTE_KEY_PREFIX, name);
+
+        return new RepositoryMetadata(name, type, buildSettings(node, settingsAttributePrefix));
```

Comment:
If there are more blobs which get added to repo metadata , we will need to add same code here as well . I am not sure how can converge `PutRepositoryRequest` with  `RemoteStoreRepositoryRegistrationHelper` in longer term .  I think we will need  to rely on the author to see where `RepositoryMetadata` is getting constructed and add it here as well .